src/LibraryManager/Manifest.cs

```diff
@@ -452,56 +453,60 @@ public void AddVersion(string version)
         private async Task<IEnumerable<FileIdentifier>> GetAllManifestFilesWithVersionsAsync(IEnumerable<ILibraryInstallationState> libraries)
         {
             var files = new List<FileIdentifier>();
+            var tasks = new List<Task<IEnumerable<FileIdentifier>>>();
 
             if (libraries != null)
             {
-                foreach (ILibraryInstallationState state in libraries.Where(l => l.IsValid(out IEnumerable<IError> errors)))
+                foreach (ILibraryInstallationState state in libraries)
                 {
-                    IProvider provider = _dependencies.GetProvider(state.ProviderId);
-
-                    if (provider != null)
-                    {
-                        ILibraryOperationResult updatedStateResult = await provider.UpdateStateAsync(state, CancellationToken.None).ConfigureAwait(false);
-
-                        if (updatedStateResult.Success)
-                        {
-                            IEnumerable<FileIdentifier> stateFiles = await GetFilesWithVersionsAsync(updatedStateResult.InstallationState).ConfigureAwait(false);
-
-                            foreach (FileIdentifier fileIdentifier in stateFiles)
-                            {
-                                if (!files.Contains(fileIdentifier))
-                                {
-                                    files.Add(fileIdentifier);
-                                }
-                            }
-                        }
-                    }
+                    tasks.Add(GetFilesWithVersionsAsync(state));
                 }
+
+                IEnumerable<FileIdentifier>[] allFiles = await Task.WhenAll(tasks).ConfigureAwait(false);
+                files = allFiles.SelectMany(f => f).Distinct().ToList();
```

Comment:
>.ToList() [](start = 62, length = 9)

nit: If files was declared as an IEnumerable above, you wouldn't need to call ToList() here.